mail/header.go

```diff
@@ -211,6 +212,61 @@ func (p *headerParser) parseMsgID() (string, error) {
 	return left + "@" + right, nil
 }
 
+func (p *headerParser) parseListCommand() (*url.URL, error) {
+	if !p.skipCFWS() {
+		return nil, errors.New("mail: malformed parenthetical comment")
+	}
+
+	// Consume a potential newline + indent.
+	p.consume('\r')
+	p.consume('\n')
+	p.skipSpace()
+
+	if p.consume('N') && p.consume('O') {
+		if !p.skipCFWS() {
+			return nil, errors.New("mail: malformed parenthetical comment")
+		}
+
+		return nil, nil
+	}
+
+	if !p.consume('<') {
+		return nil, errors.New("mail: missing '<' in list command")
+	}
+
+	i := 0
+	for p.s[i] != '>' && i+1 < len(p.s) {
+		i += 1
+	}
+
+	var lit string
+	lit, p.s = p.s[:i], p.s[i:]
+
+	u, err := url.Parse(lit)
```

Comment:
The RFC recommends that we remove any whitespace character from the string in-between the angle brackets.